### ISSUE TITLE
PHP Fatal error:  Class 'GraphQL\Examples\Blog\Data\DataSource' not found in ... graphql\graphql.php on line 27

### ISSUE THREAD

**Issue by galileomd** [webonyx/graphql-php#164]
kinda new to this - running this on a local server... 
ran composer to install library, copied graphql from examples folder to root and tried to load it from browser... (not localhost).  that's the error i keep getting.  any ideas what im doing wrong?

**Comment by vladar** [comment]
You shouldn't copy examples folder to root. Just run them from their location.

But if you copied them to root, you must also:

1. Adjust that [this line](https://github.com/webonyx/graphql-php/blob/master/examples/01-blog/graphql.php#L4) and make sure that it actually points to composer autoloader.

2. Adjust [example namespace](https://github.com/webonyx/graphql-php/blob/master/composer.json#L32) and make sure it points to the correct folder

**Comment by galileomd** [comment]
i've updated the composer accordingly so that loads correctly, but since i'm running this from a remote server i'm not able to start a php session from inside the examples folder - this is the tree structure now:
/graphql.php
   -vendor
        -webonyx
              -graphql-php
                     composer.json
                      -examples
                             -01-blog
                                     -Blog

so does the psr-4 declaration still stay as this?
      "GraphQL\\Examples\\Blog\\": "examples/01-blog/Blog/"

**Comment by vladar** [comment]
The first version seems correct (`examples/01-blog/Blog/`), but I can't help debug it remotely. This issue is related to composer project setup and doesn't relate to the library itself.

To try these examples you can always deploy them locally:
```
git clone https://github.com/webonyx/graphql-php.git .
cd graphql-php/examples/01-blog
php -S localhost:8080 ./graphql.php
```

**Comment by galileomd** [comment]
looks like there is another composer.json in the root directory.  
adding the psr-4 namespaces there made it work :/

**Comment by Chrisdowson** [comment]
@galileomd I have the some problem,How did you solve it?

**Comment by SimonMacIntyre** [comment]
I have followed this discussion and still cannot get it working. I have an empty project just to play with graphql. I don't get the expected way to use these examples. 

Neither copying a graphql to root, nor running from the vendor example folder works. In both, I adjusted paths to ensure vendor autoloader was loading, then it cannot see the datasource. 

I get that it has something to do with composer autoload paths, but I do not have the knowledge to fix the paths.
